cmd/agent/app/check.go

```diff
@@ -375,6 +364,20 @@ var checkCmd = &cobra.Command{
 			}
 		}
 
+		// only run JMX configs
+		for idx := range allConfigs {
```

Comment:
I think this might've been semi-broken all along. When we call `RunJmxListWithMetrics()`, `loadConfigs(runner)` is run here: https://github.com/DataDog/datadog-agent/pull/5229/files#diff-d82ed16123c9b4778d528606478bea94L192 and so all the relevant instances should get scheduled for collection. JMXFetch will pull and collect those when it comes up if I'm not mistaken. We might not need to run `RunJmxListWithMetrics()` multiple times.